File: src/main/java/io/vertigo/shiny/components/text/title/ShinyTitle.java
package io.vertigo.shiny.components.text.title;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyTitle(
		String title,
		int level) implements ShinyComponent {

	@Override
	public String type() {
		return "title";
	}


	public ShinyTitle {
		Assertion.check().isNotBlank(title, "Title cannot be blank");
	}
}

File: banshee/src/models/ShinyTitle.ts
import { ShinyComponent } from '../ShinyComponent';

export class ShinyTitle implements ShinyComponent {
  title: string;
  level: number;
  type: string = 'title';

  constructor(title: string, level: number) {
    this.title = title;
    this.level = level;
  }
}
